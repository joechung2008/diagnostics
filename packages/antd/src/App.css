html,
body {
  height: 100%;
  margin: 0;
  padding: 0;
}

*,
*::after,
*::before {
  box-sizing: border-box;
}

.centered {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
}

.extension-root {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
}

.flexbox {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.grow {
  flex: 1;
}

.stack {
  display: flex;
  flex-direction: row;
  height: 100%;
}

.tab-list {
  padding-left: 1rem;
}

.tab-panel {
  flex: 1;
  overflow-y: auto;
}
