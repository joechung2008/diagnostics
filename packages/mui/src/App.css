html,
body {
  height: 100%;
  margin: 0;
  padding: 0;
}

.extension-root {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  height: 100%;
  overflow-y: auto;
  padding: 0.25rem;
}

.flexbox {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.grow {
  flex: 1;
}

.stack {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  height: 100%;
  width: 100%;
}

.tab-panel {
  box-sizing: border-box;
  flex: 1;
  overflow-y: auto;
}
