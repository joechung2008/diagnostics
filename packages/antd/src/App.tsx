import { Empty, Flex, Menu, Spin, Tabs } from "antd";
import { Activity, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo, when } from "./utils";

import "./App.css";

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

type EnvironmentUrl = (typeof Environment)[keyof typeof Environment];

const items = [
  { key: "extensions", label: "Extensions" },
  { key: "build", label: "Build Information" },
  { key: "server", label: "Server Information" },
];

const App: React.FC = () => {
  const [extension, setExtension] = useState<ExtensionInfo>();
  const [environment, setEnvironment] = useState<EnvironmentUrl>(
    Environment.Public
  );
  const [selectedTab, setSelectedTab] = useState("extensions");

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR<Diagnostics>(environment, async (environment: string) => {
    const response = await fetch(environment);

    if (!response.ok) {
      throw new Error(`Error fetching diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const environmentName = useMemo(() => {
    switch (environment) {
      case Environment.Public:
        return "Public Cloud";
      case Environment.Fairfax:
        return "Fairfax";
      case Environment.Mooncake:
        return "Mooncake";
      default:
        return "Select environment";
    }
  }, [environment]);

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const environments = useMemo(
    () => [
      {
        key: "public",
        text: "Public Cloud",
        selected: environment === Environment.Public,
        onClick: () => {
          setEnvironment(Environment.Public);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "fairfax",
        text: "Fairfax",
        selected: environment === Environment.Fairfax,
        onClick: () => {
          setEnvironment(Environment.Fairfax);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "mooncake",
        text: "Mooncake",
        selected: environment === Environment.Mooncake,
        onClick: () => {
          setEnvironment(Environment.Mooncake);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
    ],
    [environment]
  );

  const menuItems = useMemo(
    () => [
      {
        key: "env-dropdown",
        label: environmentName,
        children: environments.map((env) => ({
          key: env.key,
          label: env.text,
          onClick: env.onClick,
        })),
      },
      ...when(showPaasServerless, {
        key: "paasserverless",
        label: "paasserverless",
        onClick: () => {
          const paasserverless = diagnostics?.extensions["paasserverless"];
          if (isExtensionInfo(paasserverless)) {
            setExtension(paasserverless);
            setSelectedTab("extensions");
          }
        },
      }),
      {
        key: "websites",
        label: "websites",
        onClick: () => {
          const websites = diagnostics?.extensions["websites"];
          if (isExtensionInfo(websites)) {
            setExtension(websites);
            setSelectedTab("extensions");
          }
        },
      },
    ],
    [environments, environmentName, showPaasServerless, diagnostics?.extensions]
  );

  const handleLinkClick = useCallback(
    (_?: React.MouseEvent, item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <div className="flexbox">
      <Menu mode="horizontal" items={menuItems} />
      <Tabs
        activeKey={selectedTab}
        className="tab-list"
        items={items}
        onChange={(key) => setSelectedTab(key)}
      />
      <Activity mode={isLoading ? "visible" : "hidden"}>
        <div className="tab-panel">
          <Flex className="centered">
            <Spin size="large" />
          </Flex>
        </div>
      </Activity>
      <Activity mode={error && !isLoading ? "visible" : "hidden"}>
        {error && (
          <div className="tab-panel">
            <Flex className="centered">
              <Empty
                description={`Error: ${error.message ?? "Unexpected error"}`}
              />
            </Flex>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "extensions"
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.extensions && (
          <div className="tab-panel">
            <Flex className="stack">
              <Extensions
                extensions={diagnostics.extensions}
                onLinkClick={handleLinkClick}
              />
              {extension && <Extension {...extension} />}
            </Flex>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "build" ? "visible" : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <BuildInfo {...diagnostics.buildInfo} />
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "server"
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <ServerInfo {...diagnostics.serverInfo} />
          </div>
        )}
      </Activity>
    </div>
  );
};

export default App;
