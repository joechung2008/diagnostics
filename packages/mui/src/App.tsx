import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import {
  Alert,
  Button,
  CircularProgress,
  Grid,
  Menu,
  MenuItem,
  Tab,
  Tabs,
  ThemeProvider,
  Toolbar,
} from "@mui/material";
import { Activity, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { useSystemTheme } from "./useSystemTheme";
import { isExtensionInfo } from "./utils";

import "@fontsource/roboto/300.css";
import "@fontsource/roboto/400.css";
import "@fontsource/roboto/500.css";
import "@fontsource/roboto/700.css";
import "./App.css";

const EnvironmentUrls = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

type EnvironmentUrl = (typeof EnvironmentUrls)[keyof typeof EnvironmentUrls];

const App: React.FC = () => {
  const theme = useSystemTheme();

  const [extension, setExtension] = useState<ExtensionInfo>();
  const [environment, setEnvironment] = useState<EnvironmentUrl>(
    EnvironmentUrls.Public
  );
  const [envMenuAnchorEl, setEnvMenuAnchorEl] = useState<null | HTMLElement>(
    null
  );

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR<Diagnostics>(environment, async (environment: string) => {
    const response = await fetch(environment);
    if (!response.ok) {
      throw new Error(`Error fetching diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const [selectedTab, setSelectedTab] = useState<string>("extensions");

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const environments = useMemo(
    () => [
      {
        key: "public",
        text: "Public Cloud",
        selected: environment === EnvironmentUrls.Public,
        onClick: () => {
          setEnvironment(EnvironmentUrls.Public);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "fairfax",
        text: "Fairfax",
        selected: environment === EnvironmentUrls.Fairfax,
        onClick: () => {
          setEnvironment(EnvironmentUrls.Fairfax);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "mooncake",
        text: "Mooncake",
        selected: environment === EnvironmentUrls.Mooncake,
        onClick: () => {
          setEnvironment(EnvironmentUrls.Mooncake);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
    ],
    [environment]
  );

  const handleEnvMenuOpen = useCallback(
    (event: React.MouseEvent<HTMLElement>) => {
      setEnvMenuAnchorEl(event.currentTarget);
    },
    []
  );

  const handleEnvMenuClose = useCallback(() => {
    setEnvMenuAnchorEl(null);
  }, []);

  const handleLinkClick = useCallback(
    (_?: React.MouseEvent, item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <ThemeProvider theme={theme}>
      <div className="flexbox">
        <Toolbar>
          <Button
            aria-controls={envMenuAnchorEl ? "env-menu" : undefined}
            aria-haspopup="true"
            endIcon={
              <ExpandMoreIcon
                sx={{
                  transform: envMenuAnchorEl
                    ? "rotate(180deg)"
                    : "rotate(0deg)",
                  transition: "transform 0.2s",
                }}
              />
            }
            onClick={handleEnvMenuOpen}
          >
            {environments.find((env) => env.selected)?.text ??
              "Select Environment"}
          </Button>
          <Menu
            id="env-menu"
            anchorEl={envMenuAnchorEl}
            open={Boolean(envMenuAnchorEl)}
            onClose={handleEnvMenuClose}
          >
            {environments.map((env) => (
              <MenuItem
                key={env.key}
                selected={env.selected}
                onClick={() => {
                  env.onClick();
                  handleEnvMenuClose();
                }}
              >
                {env.text}
              </MenuItem>
            ))}
          </Menu>
          <Activity mode={showPaasServerless ? "visible" : "hidden"}>
            <Button
              key="paasserverless"
              onClick={() => {
                const paasserverless =
                  diagnostics?.extensions["paasserverless"];
                if (isExtensionInfo(paasserverless)) {
                  setExtension(paasserverless);
                  setSelectedTab("extensions");
                }
              }}
            >
              paasserverless
            </Button>
          </Activity>
          <Button
            key="websites"
            onClick={() => {
              const websites = diagnostics?.extensions["websites"];
              if (isExtensionInfo(websites)) {
                setExtension(websites);
                setSelectedTab("extensions");
              }
            }}
          >
            websites
          </Button>
        </Toolbar>
        <Tabs
          value={selectedTab}
          onChange={(_, newValue) => setSelectedTab(newValue as string)}
          aria-label="App navigation"
        >
          <Tab label="Extensions" value="extensions" />
          <Tab label="Build Information" value="build" />
          <Tab label="Server Information" value="server" />
        </Tabs>
        <Activity mode={!error && isLoading ? "visible" : "hidden"}>
          <Grid className="tab-panel" container>
            <CircularProgress aria-label="Loading..." />
          </Grid>
        </Activity>
        <Activity mode={error && !isLoading ? "visible" : "hidden"}>
          <Grid className="tab-panel" container>
            <Alert severity="error">
              Error loading diagnostics: {String(error)}
            </Alert>
          </Grid>
        </Activity>
        <Activity
          mode={
            !error && !isLoading && selectedTab === "extensions"
              ? "visible"
              : "hidden"
          }
        >
          {diagnostics?.extensions && (
            <Grid className="tab-panel" container>
              <Grid className="stack" container gap="0.5rem">
                <Extensions
                  extensions={diagnostics.extensions}
                  onLinkClick={handleLinkClick}
                />
                {extension && <Extension {...extension} />}
              </Grid>
            </Grid>
          )}
        </Activity>
        <Activity
          mode={
            !error && !isLoading && selectedTab === "build"
              ? "visible"
              : "hidden"
          }
        >
          {diagnostics?.buildInfo && (
            <div className="tab-panel">
              <BuildInfo {...diagnostics.buildInfo} />
            </div>
          )}
        </Activity>
        <Activity
          mode={
            !error && !isLoading && selectedTab === "server"
              ? "visible"
              : "hidden"
          }
        >
          {diagnostics?.serverInfo && (
            <div className="tab-panel">
              <ServerInfo {...diagnostics.serverInfo} />
            </div>
          )}
        </Activity>
      </div>
    </ThemeProvider>
  );
};

export default App;
