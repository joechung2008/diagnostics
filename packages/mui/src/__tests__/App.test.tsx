import { createTheme } from "@mui/material";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import App from "../App";

// Mock useSWR
const mockUseSWR = vi.fn();
vi.doMock("swr", () => ({
  default: mockUseSWR,
}));

// Mock useSystemTheme
vi.doMock("../useSystemTheme", () => ({
  useSystemTheme: vi.fn(() => createTheme()),
}));

const diagnostics = {
  buildInfo: { buildVersion: "1.0.0" },
  extensions: {
    ext1: { extensionName: "ext1" },
  },
  serverInfo: {
    deploymentId: "deploy-1",
    extensionSync: { totalSyncAllCount: 1 },
    hostname: "localhost",
    nodeVersions: "v18.0.0",
    serverId: "server-1",
    uptime: 100,
  },
};

describe("App", () => {
  it("matches snapshot in loading state", () => {
    mockUseSWR.mockReturnValue({
      data: undefined,
      error: undefined,
      isLoading: true,
    });

    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot in error state", () => {
    mockUseSWR.mockReturnValue({
      data: undefined,
      error: new Error("Failed to fetch"),
      isLoading: false,
    });

    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot in initial loaded state", () => {
    mockUseSWR.mockReturnValue({
      data: diagnostics,
      error: undefined,
      isLoading: false,
    });
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
