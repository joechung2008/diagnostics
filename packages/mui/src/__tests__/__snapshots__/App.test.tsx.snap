// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches snapshot in error state 1`] = `
<DocumentFragment>
  <div
    class="flexbox"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
    >
      <button
        aria-haspopup="true"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Public Cloud
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-n5hn8a-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        style="display: none;"
        tabindex="0"
        type="button"
      >
        paasserverless
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        websites
      </button>
    </div>
    <div
      class="MuiTabs-root css-jhczcq-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          aria-label="App navigation"
          class="MuiTabs-list MuiTabs-flexContainer css-hzcega-MuiTabs-list"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            Extensions
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Build Information
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Server Information
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1qltlow-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row tab-panel css-d64g0g-MuiGrid-root"
    >
      <span
        aria-label="Loading..."
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row tab-panel css-d64g0g-MuiGrid-root"
      style="display: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-3lhuho-MuiPaper-root-MuiAlert-root"
        role="alert"
        style="--Paper-shadow: none;"
      >
        <div
          class="MuiAlert-icon css-vab54s-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-zioonp-MuiAlert-message"
        >
          Error loading diagnostics: undefined
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App > matches snapshot in initial loaded state 1`] = `
<DocumentFragment>
  <div
    class="flexbox"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
    >
      <button
        aria-haspopup="true"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Public Cloud
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-n5hn8a-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        style="display: none;"
        tabindex="0"
        type="button"
      >
        paasserverless
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        websites
      </button>
    </div>
    <div
      class="MuiTabs-root css-jhczcq-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          aria-label="App navigation"
          class="MuiTabs-list MuiTabs-flexContainer css-hzcega-MuiTabs-list"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            Extensions
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Build Information
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Server Information
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1qltlow-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row tab-panel css-d64g0g-MuiGrid-root"
    >
      <span
        aria-label="Loading..."
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row tab-panel css-d64g0g-MuiGrid-root"
      style="display: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-3lhuho-MuiPaper-root-MuiAlert-root"
        role="alert"
        style="--Paper-shadow: none;"
      >
        <div
          class="MuiAlert-icon css-vab54s-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-zioonp-MuiAlert-message"
        >
          Error loading diagnostics: undefined
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App > matches snapshot in loading state 1`] = `
<DocumentFragment>
  <div
    class="flexbox"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
    >
      <button
        aria-haspopup="true"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Public Cloud
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-n5hn8a-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        style="display: none;"
        tabindex="0"
        type="button"
      >
        paasserverless
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        websites
      </button>
    </div>
    <div
      class="MuiTabs-root css-jhczcq-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          aria-label="App navigation"
          class="MuiTabs-list MuiTabs-flexContainer css-hzcega-MuiTabs-list"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            Extensions
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Build Information
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1usuzwp-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Server Information
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1qltlow-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row tab-panel css-d64g0g-MuiGrid-root"
    >
      <span
        aria-label="Loading..."
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row tab-panel css-d64g0g-MuiGrid-root"
      style="display: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-3lhuho-MuiPaper-root-MuiAlert-root"
        role="alert"
        style="--Paper-shadow: none;"
      >
        <div
          class="MuiAlert-icon css-vab54s-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-zioonp-MuiAlert-message"
        >
          Error loading diagnostics: undefined
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
